### PR TITLE
Bevindingen r02 klantaanvullenmailtel

### PR DESCRIPTION
**Pull request by felixcicatt**

### CODE REVIEW

File: src/components/non-blocking-forms/NonBlockingErrors.vue [hunk 1]
```diff
@@ -11,6 +11,7 @@
 import { useValidationMessages } from "./useValidationMessages";
 
 const props = defineProps<{
+  showMessages?: boolean;
```
Comment (lencodes) [inline src/components/non-blocking-forms/NonBlockingErrors.vue:14]: This new prop is never filled in any of the implementations it seems. 

Therefore, it seems redundant.

File: src/helpers/validation.ts [hunk 2]
```diff
@@ -8,5 +8,5 @@ function isValidPhoneNumber(val: string) {
 export function customPhoneValidator(value: string): string[] {
   return isValidPhoneNumber(value)
     ? []
-    : [`Het telefoonnummer '${value}' lijkt geen valide telefoonnummer.`];
+    : [`'${value}' lijkt geen valide telefoonnummer.`];
```
Comment (lencodes) [inline src/helpers/validation.ts:11]: Why is this actually an array? Might be easier if it was a return type of `:string | undefined` and just return the error message if there's an validation error.

File: src/main.ts [hunk 3]
```diff
@@ -50,3 +50,27 @@ app.directive("focus", {
     el.stop();
   },
 });
+
+// standard html email validation is not sufficient
+const emailregex =
+  /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```
Comment (lencodes) [inline src/main.ts:56]: I would assume that a browser holds a more robust regex for validating e-mails than that this probably is. Feels as if we're trying to reinvent the wheel here and it seems hard to manage (I have no clue what this regex says, to be fair I know little to none about regex).

Above being said, it does seem to do what is requested in the test feedback.

**Review by lencodes — CHANGES_REQUESTED** [review]
Nice 👍🏼 

Left some small comments.

File: src/helpers/validation.ts (hunk 2 above)
Comment (felixcicatt) [reply src/helpers/validation.ts:11]: Just for extendibility. At this time, we just use a single validation message. In the future, we might want to give more descriptive messages (for example, a message about the length, as well as a message about strange characters). By using an array, the contract doesn't have to change in that case. Then again, we don't need it right now. I'll think about it for a bit longer.

File: src/components/non-blocking-forms/NonBlockingErrors.vue (hunk 1 above)
Comment (felixcicatt) [reply src/components/non-blocking-forms/NonBlockingErrors.vue]: Fair enough. I was a bit lazy with removing the template code. I cleaned it up now.

**Pull request merged**